site/zenodo_rdm/legacy/requests/utils.py

```diff
@@ -107,3 +108,32 @@ def submit_community_manage_record_request(user_id, uow, comment=None):
     return current_requests_service.execute_action(
         system_identity, request_item._request.id, "submit", data=comment, uow=uow
     )
+
+
+@unit_of_work()
+def submit_join_as_subcommunity_request(
+    parent_community_uuid, community_uuid, uow, comment=None
+):
+    """Create and submit a SubCommunityInvitation request."""
+    type_ = current_request_type_registry.lookup(ZenodoSubCommunityInvitationRequest.type_id)
+    creator = ResolverRegistry.resolve_entity_proxy(
+        {"community": parent_community_uuid}
+    ).resolve()
+    receiver = ResolverRegistry.resolve_entity_proxy(
+        {"community": community_uuid}
+    ).resolve()
+    expires_at = datetime.utcnow() + timedelta(days=30)
+
+    request_item = current_requests_service.create(
+        system_identity,
+        {},
+        type_,
+        receiver,
+        creator=creator,
+        expires_at=expires_at,
+        uow=uow,
+    )
```

Comment:
```suggestion
    parent_community = community_service.record_cls.pid.resolve(parent_community_id)
    child_community = community_service.record_cls.pid.resolve(child_community_id)

    expires_at = datetime.utcnow() + timedelta(days=30)

    request_item = current_requests_service.create(
        system_identity,
        {},
        type_,
        creator=parent_community,
        receiver=child_community,
        expires_at=expires_at,
        uow=uow,
    )
```

nit: initializing with the resolver and resolving is a bit of a roundtrip vs. just resolving the communities. otherwise, some renames for being explicit/descriptive about who is the creator/receiver.